internal/packages/packagecontent/topackage_multicomponent_test.go

```diff
@@ -0,0 +1,55 @@
+package packagecontent_test
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"package-operator.run/internal/packages/packagecontent"
+
+	"package-operator.run/internal/packages"
+
+	"github.com/stretchr/testify/require"
+
+	"package-operator.run/internal/packages/packageimport"
+)
+
+type testData struct {
+	directory string
+	component string
+	error     error
+}
+
+func TestMultiComponentLoader(t *testing.T) {
+	t.Parallel()
+
+	tests := []testData{
+		{"components-disabled", "", nil},
+		{"components-disabled", "foobar", packages.ViolationError{Reason: packages.ViolationReasonComponentsNotEnabled}},
+		{"components-enabled", "", nil},
+		{"components-enabled", "backend", nil},
+		{"components-enabled", "frontend", nil},
+		{"components-enabled", "foobar", packages.ErrManifestNotFound},
+	}
+
+	for i, tst := range tests {
+		test := tst
```

Comment:
nit: saves one allocation :D
```suggestion
	for i := range tests {
		test := tests[i]
```

Reply:
changed following your suggestion 👍